qiskit/algorithms/state_fidelities/base_state_fidelity.py

```diff
@@ -0,0 +1,296 @@
+        values (first and second). This calculation depends on the particular
+        fidelity method implementation, but always represents:
+
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`
+
+        where :math:`x` and :math:`y` are optional parametrizations of the
+        states :math:`\psi` and :math:`\phi` prepared by the circuits
+        ``circuit_1`` and ``circuit_2``, respectively.
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the first set of circuits
+            values_2: Numerical parameters to be bound to the second set of circuits.
+            run_options: Backend runtime options used for circuit execution.
+
+        Returns:
+            The result of the fidelity calculation.
+        """
+        raise NotImplementedError
+
+    def run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> AlgorithmJob:
+        r"""
+        Run asynchronously the state overlap (fidelity) calculation between two
+        (parametrized) circuits (first and second) for a specific set of parameter
+        values (first and second).This calculation depends on the particular
+        fidelity method implementation, but always represents:
+
+            :math:`|\langle\psi(x)|\phi(y)\rangle|^2`
+
+        where :math:`x` and :math:`y` are optional parametrizations of the
+        states :math:`\psi` and :math:`\phi` prepared by the circuits
+        ``circuit_1`` and ``circuit_2``, respectively.
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the first set of circuits.
+            values_2: Numerical parameters to be bound to the second set of circuits.
+            run_options: Backend runtime options used for circuit execution.
+
+        Returns:
+            Primitive job for the fidelity calculation.
+            The job's result is an instance of ``StateFidelityResult``.
+        """
+
+        job = AlgorithmJob(self._run, circuits_1, circuits_2, values_1, values_2, **run_options)
+
+        job.submit()
+        return job
+
+    def _truncate_fidelities(self, fidelities: Sequence[float]) -> Sequence[float]:
+        """
+        Ensure fidelity result in [0,1]
+
+        Args:
+           fidelities: sequence of raw fidelity results
+
+        Returns:
+             List of truncated fidelities.
+
+        """
+        return [0 if f < 0 else 1 if f > 1 else f for f in fidelities]
```

Comment:
This is not a strong opinion, but you can slightly simplify the code with numpy.
```suggestion
        return np.clip(fidelities, 0, 1).tolist()
```